setup.py

```diff
@@ -1,31 +1,35 @@
-# Always prefer setuptools over distutils
-from setuptools import setup, find_packages
-
 import pathlib
+from setuptools import setup, find_packages
+import sys
 
 here = pathlib.Path(__file__).parent.resolve()
 
 # Get the long description from the README file
 long_description = (here / "README.md").read_text(encoding="utf-8")
 
+ray_deps = ["ray[default]>=1.4.0", "pandas"]
+dask_deps = ["dask[complete]>=2.22.0", "distributed>=2.22.0"]
+mpi4py_deps = ["pandas", "msgpack"]
+if sys.version_info[1] < 8:
+    mpi4py_deps += "pickle5"
+all_deps = ray_deps + dask_deps + mpi4py_deps
+
 setup(
     name="unidist",
     version="0.0.1",
-    description="Unified Distributed Execution",  # Optional
-    long_description=long_description,  # Optional
-    long_description_content_type="text/markdown",  # Optional (see note above)
-    author="unidist team",  # Optional
-    author_email="author@example.com",  # Optional
-    # Classifiers help users find your project by categorizing it.
-    # For a list of valid classifiers, see https://pypi.org/classifiers/
-    classifiers=[  # Optional
-        # How mature is this project? Common values are
-        #   3 - Alpha
-        #   4 - Beta
-        #   5 - Production/Stable
-        "Development Status :: 3 - Alpha",
-        "Programming Language :: Python :: 3",
-    ],
-    packages=find_packages(),  # Required
-    python_requires=">=3.6",
+    description="Unified Distributed Execution",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    license="Apache 2",
+    packages=find_packages(),
+    url="https://github.com/modin-project/unidist",
+    install_requires=["packaging", "cloudpickle"],
+    extras_require={
+        # can be installed by pip install unidist[ray]
+        "ray": ray_deps,
+        "dask": dask_deps,
+        "mpi4py": mpi4py_deps,
```

Comment:
Possible, `mpi` naming would be better, because it's shorter and more clear for user.
```suggestion
        "mpi": mpi4py_deps,
```

Reply:
agree, changed

Reply:
removed it at all, notes on installing deps for the backend will be added to docs in a separate PR.

Reply:
Albeit we can install some deps ourselves and a user needs only to install mpi4py from conda. @prutskov , @no-ponomarev , thoughts?

Reply:
left the last one approach for now

Reply:
I agree with the last one. But we should throw warning to user with request to installing of `mpi4py` from conda channels. Is it possible to include in setup.py/requirements.txt this logic?

Reply:
Not sure that is possible. I think the notes should be enough in docs.

Reply:
In case user got an import error when he starts using MPI backend, the first entry point for user will be our github issues instead of docs I think. If it isn't possible in `setup.py`, let's add a friendly error-message instead of import error in case of using MPI backend. Good place for this is `initialize_mpi` function which is the entry point of execution

Reply:
That friendly error message might be good solution for now but I don't think `initialize_mpi` is a good place to provide the error message. I suggest merging the PR and discussing the issue in #25 or in the PR related to it.

Reply:
The change related to #25 isn't look as a huge. Also, it Is closely connected with current #22. And I don't see the barriers to resolve this as a part of the current issue. But ok, let's merge this to unlock other issues.